Source/WebCore/page/EventCounts.cpp

```diff
@@ -25,14 +25,134 @@
 
 #include "config.h"
 #include "EventCounts.h"
-
+#include "IDLTypes.h"
 #include "JSDOMMapLike.h"
+#include <algorithm>
+#include <initializer_list>
+#include <wtf/NeverDestroyed.h>
 
 namespace WebCore {
 
-void EventCounts::initializeMapLike(DOMMapAdapter&)
+static constexpr std::array COUNTED_EVENT_SET {
+    EventType::auxclick,
+    EventType::click,
+    EventType::contextmenu,
+    EventType::dblclick,
+    EventType::mousedown,
+    EventType::mouseenter,
+    EventType::mouseleave,
+    EventType::mouseout,
+    EventType::mouseover,
+    EventType::mouseup,
+    EventType::pointerover,
+    EventType::pointerenter,
+    EventType::pointerdown,
+    EventType::pointerup,
+    EventType::pointercancel,
+    EventType::pointerout,
+    EventType::pointerleave,
+    EventType::gotpointercapture,
+    EventType::lostpointercapture,
+    EventType::touchstart,
+    EventType::touchend,
+    EventType::touchcancel,
+    EventType::keydown,
+    EventType::keypress,
+    EventType::keyup,
+    EventType::beforeinput,
+    EventType::input,
+    EventType::compositionstart,
+    EventType::compositionupdate,
+    EventType::compositionend,
+    EventType::dragstart,
+    EventType::dragend,
+    EventType::dragenter,
+    EventType::dragleave,
+    EventType::dragover,
+    EventType::drop
+};
+
+EventCounts::EventCounts()
+{ }
+
+void EventCounts::add(EventType t)
+{
+    auto index = eventTypeIndex(t);
+    ASSERT(index.has_value());
+    ASSERT(index.value() < m_counts.size());
+    ++m_counts[index.value()];
+}
+
+std::optional<size_t> EventCounts::eventTypeIndex(EventType t)
+{
+    auto it = std::ranges::find(COUNTED_EVENT_SET, t);
+    if (COUNTED_EVENT_SET.end() == it)
+        return std::nullopt;
+
+    return std::distance(COUNTED_EVENT_SET.begin(), it);
+}
+
+bool EventCounts::IsCounted(EventType t)
+{
+    return !eventNameFromType(t).isNull();
+}
+
+const AtomString& EventCounts::eventNameFromType(EventType t)
+{
+    static const WTF::NeverDestroyed map(std::map<EventType, const AtomString>({
+        { EventType::auxclick           , eventNames().auxclickEvent           },
```

Comment:
This and `COUNTED_EVENT_SET` should be auto-generated from EventNames.json (see make-event-names.py)